src/plugins/prepare.rs

```diff
@@ -324,37 +331,32 @@ pub fn init_transforms<C: Component>(
                     Rotation(parent_rot.0 * rot.0)
                 }
             } else {
-                global_transform.map_or(Rotation::default(), |t| {
-                    t.compute_transform().rotation.into()
-                })
+                transform
+                    .map(|t| Rotation::from(t.rotation))
+                    .unwrap_or_else(|| {
+                        global_transform.map_or(Rotation::default(), |t| {
+                            t.compute_transform().rotation.into()
+                        })
+                    })
             }
         } else {
             default()
         };
 
-        if !config.transform_to_position {
-            return;
-        }
+        let mut cmds = commands.entity(entity);
 
         // Insert the position and rotation.
         // The values are either unchanged (Position and Rotation already exist)
         // or computed based on the GlobalTransform.
-        // If the entity isn't a rigid body, adding PreviousPosition and PreviousRotation
-        // is unnecessary.
-        if is_rb {
-            commands.entity(entity).try_insert((
-                Position(new_position),
-                *previous_pos.unwrap_or(&PreviousPosition(new_position)),
-                new_rotation,
-                *previous_rot.unwrap_or(&PreviousRotation(new_rotation)),
-                transform.map_or(Transform::default(), |t| *t),
-            ));
-        } else {
-            commands.entity(entity).try_insert((
-                Position(new_position),
-                new_rotation,
-                transform.map_or(Transform::default(), |t| *t),
-            ));
+        cmds.try_insert((
+            Position(new_position),
+            new_rotation,
+            *previous_pos.unwrap_or(&PreviousPosition(new_position)),
+            *previous_rot.unwrap_or(&PreviousRotation(new_rotation)),
+        ));
+
+        if let Some(transform) = new_transform {
+            cmds.try_insert(transform);
```

Comment:
I added the suggestion in 5febd806f78315393f62201c719865fe98d8a720.

Regarding the `RigidBody` check, should the query filter be `Has<RigidBody>` instead of `Has<C>`? Afaik, `Has<C>` has no effect in the presence of `Added<C>` (and it's not necessarily a `RigidBody` either)

Reply:
Yeah, it should be `Has<RigidBody>`. I think it was originally like that, but when I made the system more generic I accidentally changed that to `Has<C>` as well

Reply:
fixed in fde2ae88f407e12c790c2c256d366050ea2de327